### DISCUSSION TITLE
Serialization of type object is not supported. (System.Data.Entity.Spatial.DbGeography)

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#2389] [Questions]
I am using <add key="CslaSerializationFormatter" value="BinaryFormatter" /> in my application which works with DbGeography.  I can attach the values to like x, y, z to fields in WPF for display and entry.  In preparation for .Net 6, I have changed the BinaryFormatter to MobileFormatter and received the above error.  I know that DbGeography is not a primitive type so I am wondering what my options are?

**Comment by kcabral817** [comment]
I read last the night, the issues involving MF and the direction.  After I left, it dawned on me the issue.  I will create a class with the managed backfields so that Automapper can map them from the complex sql object to CSLA.

**Reply by rockfordlhotka** [reply]
That is a solution. In the future we may have a better answer.

Another current solution is to integrate a 'wrapper" into your object graph, instead of the DbGeography itself. Sort of like

```c#
[Serializable]
public class MyGeography : BusinessBase<MyGeography>
{
      public static readonly PropertyInfo<DbGeography> GeographyProperty = RegisterProperty<DbGeography>(nameof(Geography), RelationshipTypes.PrivateField);
      private DbGeography _name = GeographyProperty.DefaultValue;
      public DbGeography Geography
      {
        get => GetProperty(t, _name);
        set => SetProperty(GeographyProperty, ref _name, value);
      }

    protected override OnGetState(....)
    {
      // copy the geography values from the stream
    }

    protected override OnSetState(....)
    {
      // copy the geography values into the stream
    }
}
```

This keeps your object graph mostly like it is, and gives you the ability to serialize/deserialize the properties of the underlying non-serializable object.
